### DISCUSSION TITLE
blog/shopware-cli-project-ci/

### DISCUSSION THREAD

**Discussion by giscus** [shyim/shyim.me#55] [Q&A]
# blog/shopware-cli-project-ci/

In this short blog post, I will show you how you can use shopware-cli to build your Shopware project and deploy it later with Deployer, Docker image whatever you want.
But first what is Shopware-CLI? Shopware-CLI is a command line application written in Go. It’s not the same application as bin/console in your Project. It’s kinda a Swiss knife for the Shopware ecosystem. Likewise, it can do a lot of things like:

https://shyim.me/blog/shopware-cli-project-ci/

**Comment by lutek** [comment]
hi, i'm following your guide with docker, but struggling with problem. My shopware instance uses s3 for filesystem, on build stage got problem with assets:install (error below). As i understand building images in CI generate assets on the s3 which is not best solution (publishing new assets without image running on live env), am i missunderstanding something? 

```
9.880 !!  Copying files for bundle: Framework
9.880 !!  07:36:46 CRITICAL  [console] Error thrown while running command "assets:install". Message: "The endpoint "///files/asset-manifest.json" is invalid." ["exception" => AsyncAws\Core\Exception\InvalidArgument { …},"command" => "assets:install","message" => "The endpoint "///files/asset-manifest.json" is invalid."]
```

**Reply by shyim** [reply]
When you can't run `asset:install` in the CI. You can disable it in a .shopware-project.yml

```
build:
  disable_asset_copy: false
```

see https://sw-cli.fos.gg/shopware-project-yml-schema/

**Reply by lutek** [reply]
:/ 
```
#14 [build 3/4] RUN cat /src/.shopware-project.docker.yml
#14 0.185 build:
#14 0.185     disable_asset_copy: false
#14 DONE 0.2s
```
```
RUN --mount=type=secret,id=composer_auth,dst=/src/auth.json \
    --mount=type=cache,target=/root/.composer \
    --mount=type=cache,target=/root/.npm \
    /usr/local/bin/entrypoint.sh shopware-cli project ci --project-config /src/.shopware-project.docker.yml /src
```

**Reply by lutek** [reply]
I mean I have tested with all the options from the shopware/docker readme, still getting errors.

**Reply by shyim** [reply]
Is there maybe a post-install-hook in the composer.json?

**Reply by lutek** [reply]
LOL thank you.

**Reply by lutek** [reply]
it seems like the custom project config (.shopware-project.docker.yml) file isn't used with command `project ci`

```
/usr/local/bin/entrypoint.sh shopware-cli --project-config /src/.shopware-project.docker.yml project ci /src
```

I tested without and works  fine.

**Reply by lutek** [reply]
https://github.com/FriendsOfShopware/shopware-cli/blob/9fbf7fdb67fcc61e0ae6958b857bb65a75c7a52e/cmd/project/ci.go#L49

**Reply by shyim** [reply]
ahh yea that should be maybe improved. can you open a issue?

**Reply by lutek** [reply]
I will do.

**Comment by amenk** [comment]
Does a shopware-cli build still require "npm install"? We figured out, that the npm install takes super long in our pipelines, currently trying to artifact-cache them in GitLab --- but was wondering if shopware-cli just would avoid that all?

EDIT: Okay, it still does it ...

**Reply by shyim** [reply]
It skips the complete bundeling when the assets are there. When not we have to install node modules if they have. My experience was that mostly it is not worth to cache

**Comment by atsimerman-avenga** [comment]
Hi, is it possible to skip the storefront build? I'm following the Shopware Frontends approach and don't need the standard storefront.

**Reply by shyim** [reply]
sounds like a good use-case. can you make a issue here https://github.com/shopware/shopware-cli/issues

**Comment by lawcontinue** [comment]


**Reply by amenk** [reply]
SPAM or wrong article?
